src/main/java/io/tiledb/spark/TileDBBatch.java

```diff
@@ -284,11 +356,23 @@ private Pair<List<List<Range>>, Class> buildRangeFromFilter(
       EqualNullSafe f = (EqualNullSafe) filter;
       int columnIndex = this.tileDBReadSchema.getColumnId(f.attribute()).get();
       ranges.get(columnIndex).add(new Range(new Pair<>(f.value(), f.value())));
+
+      // for qc
+      if (arraySchema.hasAttribute(filterReferences[0])) {
```

Comment:
Are we always guaranteed that `filterReferences[0]` references the correct attribute?